src/main/java/org/takes/http/BkBasic.java

```diff
@@ -115,4 +120,23 @@ private static Response failure(final Throwable err) {
         );
     }
 
+    /**
+     * Adds custom headers with information about socket.
+     * @param req Request
+     * @param socket Socket
+     * @return Request with custom headers
+     */
+    private static Request addSocketHeaders(final Request req,
+        final Socket socket) {
+        return new RqWithHeaders(
+            req, new StringBuilder().append("X-Takes-LocalAddress: ")
+            .append(socket.getLocalAddress().toString())
+            .append("X-Takes-LocalPort")
+            .append(Integer.toString(socket.getLocalPort()))
+            .append("X-Takes-RemoteAddress")
+            .append(socket.getInetAddress().toString())
+            .append("X-Takes-RemotePort")
+            .append(Integer.toString(socket.getPort()))
```

Comment:
brrr... are you sure that how it should be? you're concatenating them all into one string... I think it should be something like:

```java
new RqWithHeaders(
  req,
  String.format("X-Takes-LocalAddress: %s", socket.getLocalAddress()),
  String.format("X-Takes-LocalPort: %d", socket.getLocalPort()),
  // etc.
)
```